### PR TITLE
fix(npmlog): use any type as message argument

### PR DESCRIPTION
**Pull request by amorscher**
- npm log uses util.format on the message object so its type should be any object

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: (https://github.com/npm/npmlog/blob/ca6b1374f3eb8261c44934cb662fd66712103d9d/lib/log.js#L237)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by amorscher** [comment]
Any help why CI is failing?

**Comment by jakebailey** [comment]
To test and see, can you (in this PR) remove the `--diff-filter` flag from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/.github/workflows/CI.yml#L57 ?

**Comment by jakebailey** [comment]
Er, no, actually, can you replace instances of `...[HEAD^1]` with `...[HEAD^1]...`? There should be two, that workflow and the azure pipelines file.

**Comment by jakebailey** [comment]
Okay, fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67239, you should be able to merge master. Thanks for noticing.

**Comment by amorscher** [comment]
Thx for the quick response !

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/npmlog/index.d.ts
1. types/npmlog/npmlog-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 72de2e5ca86b8627ab9ec0efe72d280a1d5f124e
</p>

**Comment by amorscher** [comment]
@sheetalkamat any news on that one?

**Comment by amorscher** [comment]
@DanielMSchmidt could you please have a look?

File: types/npmlog/index.d.ts [hunk 1]
```diff
@@ -15,17 +15,29 @@ declare namespace npmlog {
         heading: string;
         stream: any; // Defaults to process.stderr
 
-        log(level: LogLevels | string, prefix: string, message: string, ...args: any[]): void;
+        /**
+         * Creates a log message
+         * @param level
+         * @param prefix
+         * @param message message of the lopg which will be formatted using utils.format()
+         * @param args additional arguments
```
Comment (gabritto) [inline types/npmlog/index.d.ts:23]: Maybe specify that those are additional arguments that are passed to `utils.format`, too?

**Review by gabritto — CHANGES_REQUESTED** [review]
This change matches the types for `util.format` from `node`, which is the function that is being called with the `message, ...args` parameters by the implementation of this package, so I think it makes sense and seems unlikely to break people, even though type checking might let slide unintended usages of these functions.

I left a comment about the jsdoc comments, after that I can approve this PR.

File: types/npmlog/index.d.ts (hunk 1 above)
Comment (amorscher) [reply types/npmlog/index.d.ts]: Done

**Comment by amorscher** [comment]
@gabritto thx a lot for the detailed review!

**Review by gabritto — APPROVED** [review]

**Comment by amorscher** [comment]
Ready to merge

**Pull request merged**